src/service/osw_confidence_service.py

```diff
@@ -44,39 +46,47 @@ def process(self, msg: QueueMessage):
             process_thread = threading.Thread(target=self.calculate_confidence, args=[confidence_request])
             process_thread.start()
         except TypeError as e:
-            print(' Type error occured')
-            print(e)
-            print(msg)
-            # Need to send failure message here.
+            self.logger.error(' Type error occured')
+            self.logger.error(e)
+            self.logger.error(msg)
 
     def calculate_confidence(self, request: ConfidenceRequest):
+        local_base_path = self.settings.get_download_folder()
         # make a directory for the request
         jobId = request.data.jobId
-        local_base_path = os.path.join(self.settings.get_download_folder(), jobId)
-        if not os.path.exists(local_base_path):
-            os.makedirs(local_base_path)
+
         osw_file_local_path = os.path.join(local_base_path, 'osw.zip')  # Assuming zip file
         self.download_single_file(request.data.data_file, osw_file_local_path)
         meta_file_local_path = os.path.join(local_base_path, 'meta.json')  # Assuming json file
         self.download_single_file(request.data.meta_file, meta_file_local_path)
         # The meta file is at `meta_file_local_path`
         # The osw file is at `osw_file_local_path`
         # insert code for calculation here.
+        metric = OSWConfidenceMetric(zip_file=osw_file_local_path)
```

Comment:
If the confidence calculation only relies on osw file, comment out the download of meta file. Its an additional performance burden to download that file.